screenpy_playwright/target.py

```diff
@@ -138,15 +146,22 @@ def in_frame(self, frame_locator: str) -> Target:
         )
         return self
 
-    def __getattr__(self, name: str) -> _Manipulation:
+    def __getattribute__(self, name: str) -> _Manipulation:
         """Convert a Playwright Locator strategy into a Manipulation."""
-        if not hasattr(Locator, name):
-            msg = f"'{name}' is not a valid Playwright Locator strategy."
-            raise AttributeError(msg)
+        if hasattr(Locator, name):
+            attr = getattr(Locator, name)
+            is_property = type(attr) is property
+            r_type = getattr(attr, "__annotations__", {}).get("return")
+
+            if is_property or r_type is Locator or r_type == "Locator":
+                manipulation = _Manipulation(self, name)
+                self.manipulations.append(manipulation)
+                return manipulation
+
+            msg = f"'{name}' cannot be accessed until `found_by` is called."
+            raise TargetingError(msg)
 
-        manipulation = _Manipulation(self, name)
-        self.manipulations.append(manipulation)
-        return manipulation
+        return super().__getattribute__(name)
```

Comment:
This is the aforementioned whizbangery.

We're using `__getattribute__` for this work because we need to make sure we don't call the `Locator` methods until `found_by` actually returns it. Instead of calling the method, we append a `_Manipulation` to the manipulations list before returning it. That `_Manipulation` will wait to see if it's called and track the args/kwargs used, or it will pretend it's a `Target` when `found_by` is called.